### DISCUSSION TITLE
Documentation issue

### DISCUSSION THREAD

**Discussion by hemeleerse** [eModbus/eModbus#226] [Q&A] [answered]
I'm looking at the documentation on Modbus constructors specifically at the one for writing multiple reqisters: 

_ModbusMessage(uint8_t serverID, uint8_t functionCode, uint16_t p1, uint16_t p2, uint8_t count, uint16_t *arrayOfWords)_

My question is about this comment :
_Another frequently used Modbus function code is WRITE_MULT_REGISTERS, 0x10. It requires a 16bit address, a numer of words to write (16bit again), a length byte and a uint8_t* pointer to an array of words to be written. That is the purpose of this constructor here._

- Why, besides the number of 16 bit words to write, a length byte is needed ? (is this the count in the constructor ? this is confusing)
- Why does the constructor uses a uint16_t pointer while the explanation mentions unint8_t ?

Some additional explanation would really help !

**Comment by Miq1** [comment, marked as answer]
That unfortunately is required by the Modbus standard an thus was taken over unreflected into the call. The register count (``uint16_t``) and length byte (``uint8_t``) are somewhat redundant indeed, as the latter will be double of the former in all cases. Moreover, given the capacity of a ``uint8t``, this will restrict the number of registers in one call to 126 in any case, although technically it does exist in Modbus RTU only.
We could have done the calculation and addition of that length byte internally, right, but decided against to not be different from the standard.
